ee/control/consumers/keyvalueconsumer/config_consumer.go

```diff
@@ -0,0 +1,47 @@
+package keyvalueconsumer
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/ee/agent/types"
+)
+
+type ConfigConsumer struct {
+	updater types.Updater
+}
+
+func NewConfigConsumer(updater types.Updater) *ConfigConsumer {
+	c := &ConfigConsumer{
+		updater: updater,
+	}
+
+	return c
+}
+
+func (c *ConfigConsumer) Update(data io.Reader) error {
+	if c == nil {
+		return errors.New("key value consumer is nil")
+	}
+
+	var kvPairs map[string]any
```

Comment:
Discovered when testing with @jbeker that the existing KV consumer in this package does not work for this config because Golang won't unmarshal the config as a `map[string]string` since it's a more complex object, which makes sense.

This is a quick fix, but maybe we want something cleverer?

Reply:
Correct, this handles the error that Jeremy and I were seeing: `failed to decode key-value json: json: cannot unmarshal object into Go value of type string`.

I wasn't sure if it would be cleaner for K2 to update or not.

Reply:
It's going to be pretty trivial on the k2 side. And I think it's pretty trivial here. 

So I think we get to decide what feels like a cleaner protocol. 

Having everything as a string _feels_ clean, but having the KATC definitions marshalled into strings feels kinda weird. 🤷

Reply:
I guess I like it this way a little better, then -- it feels tidy that k2 doesn't need to know about the marshalling and unmarshalling to a string; launcher handles that itself for ease of storage in bucket/startup_settings.

Reply:
It's step towards a more structured interface and API between k2 and launcher. (vs what we have now, which feels like it requires a very complete knowledge of all the parts)